src/App.tsx

```diff
@@ -1,24 +1,190 @@
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    })
+      .then((result) => {
+        setTodos(prevTodos => [...prevTodos, result]);
+      })
+      .catch(() => {
+        setVisibleError('Unable to add a todo');
+      })
+      .finally(() => {
+        setIsTodoLoading(false);
+        setTempTodo(null);
+        setTodoTitle('');
+      });
+  }, [todoTitle, todos]);
+
+  const removeTodo = (todoId: number) => {
+    setDeletedTodoId(prevState => prevState.filter(id => id !== todoId));
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(todos.filter(todo => todo.id !== todoId));
+      })
+      .catch(() => {
+        setVisibleError('Unable to delete a todo');
+      });
+  };
+
+  const handleToggleButton = async () => {
+    try {
+      const updatedTodos = todos.map(todo => ({
+        ...todo,
+        completed: false,
+      }));
+
+      setTodos(updatedTodos);
+
+      await Promise.all(
+        updatedTodos.map(todo => updateTodo(todo.id, todo)),
+      );
+    } catch (error) {
+      setVisibleError('Unable to update todos');
+    }
+  };
+
+  const handleCheck = async (todoId: number) => {
+    try {
+      const updatedTodos = todos.map(todo => {
+        if (todo.id === todoId) {
+          return { ...todo, completed: !todo.completed };
+        }
+
+        return todo;
+      });
+
+      setTodos(updatedTodos);
+
+      await Promise.all(
+        updatedTodos.map(todo => updateTodo(todo.id, todo)),
+      );
+    } catch (error) {
+      setVisibleError('Unable to update todo');
+    }
+  };
+
+  const removeCompletedTodos = useCallback(() => {
+    const completedTodoIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    Promise.all(
+      completedTodoIds.map(id => (
+        client.delete(`/todos/${id}`)
+          .catch(() => {
+            setVisibleError(`Unable to delete todo with ID ${id}`);
+          }))),
+    )
+      .then(() => {
+        const filteredTodos = todos.filter(todo => !todo.completed);
+
+        setTodos(filteredTodos);
+      })
+      .catch(() => {
+        setVisibleError('Unable to delete completed todos');
+      });
+  }, [todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <Header
+        todos={todos}
+        handleFormSubmit={handleFormSubmit}
+        todoTitle={todoTitle}
+        setTodoTitle={setTodoTitle}
+        isTodoLoading={isTodoLoading}
+        handleToggleButton={handleToggleButton}
+      />
```

Comment:
In my opinion, you can pass only todos and two functions, and then write all the logic inside the component